### DISCUSSION TITLE
Looking for API logs

### DISCUSSION THREAD

**Discussion by jharris1679** [gnzsnz/ib-gateway-docker#160] [Q&A]
IB documetnation suggests API logs should be in in a folder that looks like Jts/acknibfem... I see the audit trail file, with a .trd extension, but should the actual API logs be in there as well? Is there a setting to be changed in the IB Gateway to ensure the logfile is created?

Thanks in advance

**Comment by gnzsnz** [comment]
this is how i get my logs

https://interactivebrokers.github.io/tws-api/support.html#api_logs

**Comment by jharris1679** [comment]
Thanks for the quick response, are you finding the `api.[clientId].[day].log` file inside your ib-gateway container? This is what I'm looking for according to those docs you shared, but I'm not able to find it. Maybe there's an environment variable I need to change in the container to enable writing logs? 
Edit: Here's what I am seeing inside of the ib-gateway container
<img width="916" alt="Screenshot 2024-09-09 at 2 06 27 PM" src="https://github.com/user-attachments/assets/c768536b-cfe6-4435-bda7-1eca66be6a91">

**Reply by gnzsnz** [reply]
what IBC settting are you looking for?

**Reply by jharris1679** [reply]
Well what I'm looking for is the log file but I'm wondering if it is not there because ib-gateway is not configured to write it, similar to the behaviour of TWS where the "Create API Message Log File" needs to be checked. Is there an equivalent in the ib-gateway? Or does it borrow from my account's settings? 

Thanks again for your eyes on this

**Reply by gnzsnz** [reply]
this docker images wraps ibgateway/TWS with IBC, it does what they do plus a few extras.

the way to get the logs is described here -> https://interactivebrokers.github.io/tws-api/support.html#api_logs

if you follow those steps, you will be able to enable logs, and then you can get your logs. make sure that you map a volume so you can write logs to a place that can be accessed from your host.

**Reply by JohnAllen** [reply]
This repo is about running TWS in a container but this is telling the TWS Gui to write the log file. If someone has figured out how to easily get the Gateway/TWS log file it would be great if you would share it. Thanks!
